### PR TITLE
[Bugfix] Fix the problem when read as a delta table

### PR DESCRIPTION
**Pull request by danpi**
Fixes [#130](https://github.com/streamnative/eng-support-tickets/issues/130)

### Test report

https://github.com/streamnative/pulsar-io-lakehouse/issues/171

### Documentation

- [x] `no-need-doc`

### CODE REVIEW

**Comment by danpi** [comment]
> I'm a little curious. I also use it to write data to the delta table. But I can read the data successfully. image

@horizonzy Yes, it behaves as expected sometimes. To describe the problem, I constructed an inevitable scenario in the test report.

The root cause is that the size currently written in the description file(xxx.json) is inconsistent with the size of the parquet file. If the amount of data is small, the bufferSize will be small, and there will be problems in reading. If 1000 pieces of data are written into a parquet file, the bufferSize will be very large, maybe 20000B, but the parquet file size may only be 4000B(because of compression). So there seems to be no problem, but it just doesn't show up.

For the reading exception scenario, you can look at the image below to get a better understanding:
image

**Comment by horizonzy** [comment]
Got it. Thanks.

**Comment by horizonzy** [comment]
Need cherry-pick to branch-2.10

File: src/main/java/org/apache/pulsar/ecosystem/io/lakehouse/parquet/DeltaParquetFileWriter.java [hunk 1]
```diff
@@ -138,12 +135,12 @@ private void close() throws IOException {
     public static String generateNextFilePath(String partitionColumnPath, String tablePath, String compression) {
         StringBuilder sb = new StringBuilder();
         String suffix = new StringBuilder()
-            .append("part-0000-")
```
Comment (horizonzy) [inline src/main/java/org/apache/pulsar/ecosystem/io/lakehouse/parquet/DeltaParquetFileWriter.java:141]: Needn't change the format

**Review by horizonzy — COMMENTED** [review]
Some code style issues, could you revert the format.

**Comment by danpi** [comment]
> Some code style issues, could you revert the format.

Done. PTAL.

**Comment by danpi** [comment]
> Need cherry-pick to branch-2.10

Does branch-2.9 also need?

**Comment by horizonzy** [comment]
> > Need cherry-pick to branch-2.10
> 
> Does branch-2.9 also need?

also need.

**Review by horizonzy — APPROVED** [review]
LGTM.

**Pull request merged**